### PR TITLE
feat: EFB Dashboard Weather Info With Full METAR

### PR DESCRIPTION
**Pull request by frankkopp**
## Summary of Changes
This PR adds the full METAR text to the flyPad dashboard weather widget. 
For this some of the layout had to be changed and a scrollbar added. 
It also remembers the chosen format between sim starts separately for origin and destination. 

## Screenshots (if necessary)
https://user-images.githubusercontent.com/16833201/149624354-6f20e4a1-d181-4b56-b5d7-072327b804ef.mp4

Discord username (if different from GitHub): Cdr_Maverick#6475

## Testing instructions

- Start anywhere
- Load EFB from simbrief or manually using the weather ICAO input fields. 
- Use the ICAO field first, before loading from simbrief. Make sure the "FROM SIMBRIEF" overwrites the manually entered ICAO. 
- Check various examples and watch for layout issues. 
- Check that it remembers the chosen format between sim starts separately for origin and destination. 

## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page

### CODE REVIEW

File: src/instruments/src/EFB/Dashboard/Widgets/WeatherWidget.tsx [hunk 1]
```diff
@@ -73,6 +73,9 @@ type WeatherWidgetProps = { name: string, editIcao: string, icao: string};
 const WeatherWidget = (props: WeatherWidgetProps) => {
     const [metar, setMetar] = useState<MetarParserType>(MetarParserTypeProp);
 
+    // const [showMetar, setShowMetar] = useState('DISABLED');
```
Comment (ErickSharp) [inline src/instruments/src/EFB/Dashboard/Widgets/WeatherWidget.tsx:76]:
```suggestion

```

File: src/instruments/src/EFB/Dashboard/Widgets/WeatherWidget.tsx [hunk 2]
```diff
@@ -115,100 +118,135 @@ const WeatherWidget = (props: WeatherWidgetProps) => {
...
+                                                            strokeLinejoin="miter"
+                                                        />
+                                                        {' '}
+                                                        /
+                                                        {' '}
+                                                        {metar.wind.speed_kts.toFixed(0)}
+                                                        {' '}
+                                                        kts
+                                                    </>
+                                                ) : 'N/A'}
```
Comment (ErickSharp) [inline src/instruments/src/EFB/Dashboard/Widgets/WeatherWidget.tsx:185]: It seems like this code segments similar to this one are recurrent throughout this file; consider extracting this into a component for conciseness

**Review by ErickSharp — CHANGES_REQUESTED** [review]

File: src/instruments/src/EFB/Dashboard/Widgets/WeatherWidget.tsx (hunk 2 above)
Comment (frankkopp) [reply src/instruments/src/EFB/Dashboard/Widgets/WeatherWidget.tsx:185]: That is basically as it was before - I just fixed a small bug by using metar.raw_text as condition to display value or N/A.
As these section are not identical but only similar it is probably best to leave them for now with you already working on v3.

File: src/instruments/src/EFB/Dashboard/Widgets/WeatherWidget.tsx [hunk 3]
```diff
@@ -115,100 +115,135 @@ const WeatherWidget = (props: WeatherWidgetProps) => {
...
+                                className="text-center w-24 ml-4 text-2xl font-medium uppercase"
+                                placeholder={props.icao}
+                                value={props.icao === '----' ? '' : props.icao}
+                                onChange={(value) => handleIcao(value)}
+                                maxLength={4}
+                            />
+                            <div className="ml-6">
+                                <button
+                                    type="button"
+                                    className="mr-1 w-24 text- bg-gray-600 p-2 flex items-center justify-center rounded-lg focus:outline-none text-lg"
```
Comment (ErickSharp) [inline src/instruments/src/EFB/Dashboard/Widgets/WeatherWidget.tsx:133]:
```
className="mr-1 w-24 text- bg-gray-600 p-2 flex items-center justify-center rounded-lg focus:outline-none text-lg"
```

`text-` is incomplete

Comment (frankkopp) [reply src/instruments/src/EFB/Dashboard/Widgets/WeatherWidget.tsx]: Fixed

**Comment by John-UK** [comment]
Quality Assurance Report

Discord : John P Maguire#9027
Object of testing: #6529 
Tier of Testing : 1
Date : 26/01/2022

Testing Process:
Loaded EFB both manually and from SIMBRIEF. METAR data was loaded and displayed as icons and text correctly. 

Entering an invalid ICAO no negative impact. 
ICAO = IATA codes as tested.  

From SIMBRIEF button overwrote manual entries on first use with correct dep arr.

Closed sim with different formats set icon-icon text-icon etc. Format was persistent on reloads.  

Observations:
Not sure where average wind came from. It didn’t match my SIMBRIEF trip wind component.

Negatives:
N/A
Testing Results:
Passed

Conclusions:
LGTM

**Comment by mico975** [comment]
Quality Assurance Trainee Report

Discord          : mico#3145
Object of testing: #6529 
Tier of Testing  : 1
Date             : 26/01/2022

Testing Process: 
Tested manual input and then overwrite by Simbrief request - OK
Requested various METAR stations - OK, no layout issues
Requested invalid ICAO and got NO VALID ICAO CHOSEN - OK

Remember format on sim restart - OK

Negatives:       
N/A

Testing Results: 
Passed

Conclusions: 
LGTM

Media:       
image

**Pull request merged**